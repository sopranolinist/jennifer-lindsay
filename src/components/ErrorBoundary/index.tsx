import {
  createStyles,
  Theme,
  withStyles,
  WithStyles
} from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import React, { Component, ReactChild } from 'react';
import DocumentTitle from 'react-document-title';
import { getPageTitle } from '../../helpers/DataHelper';

const styles = (theme: Theme) =>
  createStyles({
    root: {
      alignItems: 'baseline',
      padding: theme.spacing(3),
      paddingTop: theme.spacing(12),
      flexGrow: 1,
      flexDirection: 'column',
      display: 'flex',
      textAlign: 'left'
    },
    errorHeader: {
      color: theme.palette.error['main'],
      marginBottom: theme.spacing(2)
    },
    errorMessage: {
      marginBottom: theme.spacing(2)
    }
  });

interface Props extends WithStyles<typeof styles> {
  children?: ReactChild;
  hasError?: string;
}
interface State {
  error: boolean;
}

class ErrorBoundary extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = { error: false };
  }

  static getDerivedStateFromError(error: Error | null) {
    return { error: true };
  }

  componentDidCatch(error: Error | null, info: object) {
    // console.log('ErrorBoundary.componentDidCatch', error, info);
  }

  render() {
    const pageTitle = 'Error';
    const { error } = this.state;
    const { children, hasError, classes } = this.props;
    if (!error && !hasError) return children;
    return (
      <div className={classes.root}>
        <DocumentTitle title={getPageTitle(pageTitle)} />
        <Typography component="h1" variant="h6" className={classes.errorHeader}>
          Error
        </Typography>
        <Typography
          component="span"
          variant="body1"
          className={classes.errorMessage}
        >
          The site has encountered an error{hasError && ':'}
        </Typography>
        {hasError && (
          <Typography
            component="span"
            variant="body1"
            className={classes.errorMessage}
          >
            {hasError}
          </Typography>
        )}
      </div>
    );
  }
}

export default withStyles(styles)(ErrorBoundary);
