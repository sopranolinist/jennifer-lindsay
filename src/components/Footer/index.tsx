import Divider from '@material-ui/core/Divider';
import Grid from '@material-ui/core/Grid';
import { createStyles, withStyles, WithStyles } from '@material-ui/core/styles';
import { Theme } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import classNames from 'classnames';
import React, { useContext } from 'react';
import { LayoutContext } from '../../hooks/LayoutContext';
import { drawerWidth } from './../../helpers/NavHelper';
const styles = (theme: Theme) =>
  createStyles({
    footer: {
      background: '#ffffff',
      height: 56,
      [theme.breakpoints.up('sm')]: {
        padding: `0 ${theme.spacing(3)}px`
      },
      [theme.breakpoints.up('md')]: {
        height: 64,
        paddingLeft: theme.spacing(12),
        paddingRight: theme.spacing(12)
      }
    },
    footerHasRightDrawer: {
      [theme.breakpoints.up('md')]: {
        paddingLeft: theme.spacing(3),
        paddingRight: theme.spacing(3)
      },
      [theme.breakpoints.up('lg')]: {
        paddingLeft: theme.spacing(12),
        paddingRight: theme.spacing(12)
      }
    },
    footerLoggedIn: {
      [theme.breakpoints.up('xl')]: {
        paddingLeft: `calc(${theme.spacing(12)}px + ${drawerWidth}px)`
      }
    },
    footerContent: {
      display: 'flex',
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'center',
      [theme.breakpoints.up('sm')]: {
        justifyContent: 'space-between'
      }
    },
    textLink: {
      color: theme.palette.primary.dark
    },
    [theme.breakpoints.up('sm')]: {
      footerContentLt: {
        justifyContent: 'flex-start'
      },
      footerContentRt: {
        justifyContent: 'flex-end'
      }
    }
  });

interface Props extends WithStyles<typeof styles> {
  version: string | undefined;
  year: number;
}

const Footer = ({ classes, version, year }: Props) => {
  const layoutContext = useContext(LayoutContext);
  if (!layoutContext) {
    throw new Error('Layout context is missing');
  }

  const { rightDrawerOpen } = layoutContext;
  return (
    <div data-testid="component-footer">
      <Divider />
      <Grid
        container
        direction="row"
        justifyContent="space-between"
        alignItems="center"
        className={classNames(classes.footer, {
          [classes.footerLoggedIn]: true,
          [classes.footerHasRightDrawer]: rightDrawerOpen
        })}
      >
        <Grid
          item
          xs={12}
          sm={10}
          className={`${classes.footerContent} ${classes.footerContentLt}`}
        >
          <Typography
            variant="subtitle2"
            component="span"
            className={classes.legal}
          >
            {`Â© ${year} `}
            <a className={classes.textLink} href="https://jenniferlindsay.com/">
              Jennifer Lindsay
            </a>
            {` | All Rights Reserved`}
          </Typography>
        </Grid>
        <Grid
          item
          xs={12}
          sm={2}
          className={`${classes.footerContent} ${classes.footerContentRt}`}
        >
          <Typography variant="subtitle2" component="span">
            {`Version ${version}`}
          </Typography>
        </Grid>
      </Grid>
    </div>
  );
};

Footer.defaultProps = {
  version: process.env.REACT_APP_VERSION,
  year: new Date().getFullYear()
};

export default withStyles(styles)(Footer);
