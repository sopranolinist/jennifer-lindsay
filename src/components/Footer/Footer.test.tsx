import { render } from '@testing-library/react';
import { advanceTo, clear } from 'jest-date-mock';
import React from 'react';
import {
  authExamples,
  getFullyWrappedComponent,
  getJssWrappedComponent,
  getRouterWrappedComponent,
  mockConsole
} from '../../helpers/TestHelper';
import ErrorBoundary from '../ErrorBoundary';
import Footer from './';

interface Props {}

describe('Component - Footer', () => {
  beforeEach(() => {
    mockConsole();
    advanceTo(1558461651876);
  });
  afterEach(() => {
    clear();
  });

  test('should enforce auth context', async done => {
    const { container } = render(
      getJssWrappedComponent(
        getRouterWrappedComponent(
          <ErrorBoundary>
            <Footer />
          </ErrorBoundary>
        )
      )
    );
    expect(container).toMatchSnapshot();
    done();
  });

  test('renders the footer screen', async done => {
    const { authContextProps, authObjProps } = authExamples.signedInAuthorized;
    const { container } = render(
      getFullyWrappedComponent(<Footer />, { authContextProps, authObjProps })
    );
    expect(container).toMatchSnapshot();
    done();
  });
});
