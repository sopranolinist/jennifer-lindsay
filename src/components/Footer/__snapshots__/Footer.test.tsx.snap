// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Footer renders the footer screen 1`] = `
<div>
  <div
    data-testid="component-footer"
  >
    <hr
      class="MuiDivider-root"
    />
    <div
      class="MuiGrid-root Footer-footer-4 Footer-footerLoggedIn-6 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
    >
      <div
        class="MuiGrid-root Footer-footerContent-7 Footer-footerContentLt-9 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-10"
      >
        <span
          class="MuiTypography-root MuiTypography-subtitle1"
        >
          Â© 2020 
          <a
            class="Footer-textLink-8"
            href="https://diamond.la/"
          >
            Diamond
          </a>
          , All rights reserved
        </span>
      </div>
      <div
        class="MuiGrid-root Footer-footerContent-7 Footer-footerContentRt-10 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-2"
      >
        <span
          class="MuiTypography-root MuiTypography-subtitle1"
        >
          Version 1.1.0
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component - Footer should enforce auth context 1`] = `
<div>
  <div
    class="ErrorBoundary-root-1"
  >
    <h1
      class="MuiTypography-root ErrorBoundary-errorHeader-2 MuiTypography-h6"
    >
      Error
    </h1>
    <span
      class="MuiTypography-root ErrorBoundary-errorMessage-3 MuiTypography-body1"
    >
      The site has encountered an error
    </span>
  </div>
</div>
`;
